sale_payment_method/sale.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Guewen Baconnier, Sébastien Beau
+#    Copyright (C) 2011 Akretion Sébastien BEAU <sebastien.beau@akretion.com>
+#    Copyright 2013 Camptocamp SA (Guewen Baconnier)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import api, models, fields, exceptions, _
+import openerp.addons.decimal_precision as dp
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.one
+    @api.depends('amount_total', 'payment_ids.credit', 'payment_ids.debit')
+    def _compute_amount(self):
+        paid_amount = sum(line.credit - line.debit
+                          for line in self.payment_ids)
+        self.amount_paid = paid_amount
+        self.residual = self.amount_total - paid_amount
+
+    @api.one
+    @api.depends('payment_ids')
+    def _compute_payment_exists(self):
```

Comment:
No, it doesn't work everywhere.

If in the view I use this:
```
                <button string="Create Invoice" position="after">
                    <field name="payment_ids" invisible="1"/>
                    <button name="action_view_payments"
                        string="View Automatic Payment"
                        type="object" class="oe_highlight"
                        attrs="{'invisible': [('payment_ids', '=', False)]}"
                        groups="base.group_user"/>
                </button>
```
Or this:
```
                <button string="Create Invoice" position="after">
                    <field name="payment_ids" invisible="1"/>
                    <button name="action_view_payments"
                        string="View Automatic Payment"
                        type="object" class="oe_highlight"
                        attrs="{'invisible': [('payment_ids', '=', [])]}"
                        groups="base.group_user"/>
                </button>
```

The button is always displayed. This has not changed since v7.